2.7.0: Test failure with guessit in test_episode_fromname_guessit_bug

Hi,

While updating/testing subliminal with newer dependencies, the test test_episode_fromname_guessit_bug fails because the latest guessit version parses the adam-12 sample differently:

video.series is now parsed as 'Adam-12' instead of 'Adam 12'.

video.title is now successfully populated with 'Log 141 The Color TV Bandit' instead of None.

Errors that I've found:

```
E       AssertionError: assert 'Adam-12' == 'Adam 12'
E
E         - Adam 12
E         ?      ^
E         + Adam-12
E         ?      ^

/wrkdirs/usr/ports/multimedia/py-subliminal/work-py312/subliminal-2.7.0/tests/test_video.py:250: AssertionError
```
and
```
E       AssertionError: assert 'Log 141 The Color TV Bandit' is None
E        +  where 'Log 141 The Color TV Bandit' = <Episode [Adam-12 (1968) s01e02]>.title

/wrkdirs/usr/ports/multimedia/py-subliminal/work-py312/subliminal-2.7.0/tests/test_video.py:253: AssertionError
```

I fixed test locally with:

```diff
--- tests/test_video.py.orig	2026-07-22 11:59:48 UTC
+++ tests/test_video.py
@@ -247,8 +247,8 @@ def test_episode_fromname_guessit_bug(episodes: dict[s
     assert video.release_group == episodes['adam-12_s01e02'].release_group
     assert video.resolution == episodes['adam-12_s01e02'].resolution
     assert video.video_codec == episodes['adam-12_s01e02'].video_codec
-    assert video.series == episodes['adam-12_s01e02'].series
+    assert video.series in ('Adam 12', 'Adam-12')
     assert video.season == episodes['adam-12_s01e02'].season
     assert video.episode == episodes['adam-12_s01e02'].episode
-    assert video.title is None
+    assert video.title in (None, 'Log 141 The Color TV Bandit')
     assert video.year == episodes['adam-12_s01e02'].year

```

Thanks

Hello, thanks for reporting.

What's the `guessit` version used? The github CI run with the latest 4.1.0 version with no error.

Anyway, maybe I should add a freebsd CI test :)

On 22 July 2026 13:20:20 GMT+01:00, Nuno Teixeira ***@***.***> wrote:
>nunotexbsd created an issue (Diaoul/subliminal#1368)
>
>Hi,
>
>While updating/testing subliminal with newer dependencies, the test test_episode_fromname_guessit_bug fails because the latest guessit version parses the adam-12 sample differently:
>
>video.series is now parsed as 'Adam-12' instead of 'Adam 12'.
>
>video.title is now successfully populated with 'Log 141 The Color TV Bandit' instead of None.
>
>Errors that I've found:
>
>```
>E       AssertionError: assert 'Adam-12' == 'Adam 12'
>E
>E         - Adam 12
>E         ?      ^
>E         + Adam-12
>E         ?      ^
>
>/wrkdirs/usr/ports/multimedia/py-subliminal/work-py312/subliminal-2.7.0/tests/test_video.py:250: AssertionError
>```
>and
>```
>E       AssertionError: assert 'Log 141 The Color TV Bandit' is None
>E        +  where 'Log 141 The Color TV Bandit' = <Episode [Adam-12 (1968) s01e02]>.title
>
>/wrkdirs/usr/ports/multimedia/py-subliminal/work-py312/subliminal-2.7.0/tests/test_video.py:253: AssertionError
>```
>
>I fixed test locally with:
>
>```diff
>--- tests/test_video.py.orig	2026-07-22 11:59:48 UTC
>+++ tests/test_video.py
>@@ -247,8 +247,8 @@ def test_episode_fromname_guessit_bug(episodes: dict[s
>     assert video.release_group == episodes['adam-12_s01e02'].release_group
>     assert video.resolution == episodes['adam-12_s01e02'].resolution
>     assert video.video_codec == episodes['adam-12_s01e02'].video_codec
>-    assert video.series == episodes['adam-12_s01e02'].series
>+    assert video.series in ('Adam 12', 'Adam-12')
>     assert video.season == episodes['adam-12_s01e02'].season
>     assert video.episode == episodes['adam-12_s01e02'].episode
>-    assert video.title is None
>+    assert video.title in (None, 'Log 141 The Color TV Bandit')
>     assert video.year == episodes['adam-12_s01e02'].year
>
>```
>
>Thanks
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Diaoul/subliminal/issues/1368
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>

--- a/changelog.d/1368.misc.rst
+++ b/changelog.d/1368.misc.rst
@@ -1,0 +1,1 @@
+fix test for compatibility with guessit>=4

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -67,6 +67,7 @@ docs = [
 tests = [
     "subliminal[rar]",
     "coverage[toml]>=7",
+    "packaging",
     "pytest>=6.0",
     "pytest-cov",
     "pytest-xdist",

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -520,12 +520,13 @@ def episodes() -> dict[str, Episode]:
         ),
         'adam-12_s01e02': Episode(
             'Adam-12 1968 Season 1 Complete x264 [i_c]/Adam-12 S01E02 Log 141 The Color TV Bandit.mkv',
-            'Adam 12',
+            'Adam-12',  # with guessit>=4, before 'Adam 12'
             1,
             2,
             year=1968,
             release_group='[i_c]',
             video_codec='H.264',
+            title='Log 141 The Color TV Bandit',  # with guessit>=4, before None
         ),
     }
 

--- a/tests/test_video.py
+++ b/tests/test_video.py
@@ -3,10 +3,12 @@ from __future__ import annotations
 
 import os
 from datetime import datetime, timedelta, timezone
+from importlib.metadata import version as get_version
 from typing import TYPE_CHECKING
 from unittest.mock import Mock
 
 import pytest
+from packaging.version import Version
 
 from subliminal.utils import sanitize, timestamp
 from subliminal.video import Episode, Movie, Video
@@ -239,8 +241,14 @@ def test_episode_fromname(episodes: dict[str, Episode]) -> None:
     assert video.tvdb_id is None
 
 
+@pytest.mark.skipif(
+    Version(get_version('guessit')) < Version('4'),
+    reason='guessit was not correctly parsing this video in older versions',
+)
 def test_episode_fromname_guessit_bug(episodes: dict[str, Episode]) -> None:
-    # Only works with Video.fromname, not Episode.fromname
+    # Only works with Video.fromname
+    # With Episode.fromname, the episode is incorrectly guessed as 12
+    # See https://github.com/guessit-io/guessit/issues/929
     video = Video.fromname(episodes['adam-12_s01e02'].name)
     assert isinstance(video, Episode)
     assert video.name == episodes['adam-12_s01e02'].name
@@ -250,5 +258,5 @@ def test_episode_fromname_guessit_bug(episodes: dict[str, Episode]) -> None:
     assert video.series == episodes['adam-12_s01e02'].series
     assert video.season == episodes['adam-12_s01e02'].season
     assert video.episode == episodes['adam-12_s01e02'].episode
-    assert video.title is None
+    assert video.title == episodes['adam-12_s01e02'].title
     assert video.year == episodes['adam-12_s01e02'].year

